libkbfs/folder_block_manager.go

```diff
@@ -1161,18 +1169,29 @@ func (fbm *folderBlockManager) doReclamation(timer *time.Timer) (err error) {
 }
 
 func (fbm *folderBlockManager) reclaimQuotaInBackground() {
-	timer := time.NewTimer(fbm.config.QuotaReclamationPeriod())
+	timer := time.NewTimer(fbm.config.Mode().QuotaReclamationPeriod())
 	timerChan := timer.C
 	for {
 		// Don't let the timer fire if auto-reclamation is turned off.
-		if fbm.config.QuotaReclamationPeriod().Seconds() == 0 {
+		if fbm.config.Mode().QuotaReclamationPeriod().Seconds() == 0 {
 			timer.Stop()
 			// Use a channel that will never fire instead.
 			timerChan = make(chan time.Time)
 		}
+
+		state := keybase1.AppState_FOREGROUND
 		select {
 		case <-fbm.shutdownChan:
 			return
+		case state = <-fbm.g.AppState.NextUpdate(&state):
+			for state != keybase1.AppState_FOREGROUND {
```

Comment:
After app is backgrounded, it can be suspended by the OS. So if a QR has started at this point, would it make sense to try to stop it if it doesn't finish within a short period? Perhaps could pass in a context with deadline into `doReclaimation()`.

Reply:
QR can't be going at this point, since the loop waits on each QR before doing the `select`.  Are you proposing having a second background goroutine that waits on the signals?  That seems a bit too complicated for this PR -- I vote that if we get backgrounded, we just let the QR finish and/or get suspended.  We can revisit later if it becomes a problem.

Reply:
Oh I didn't realize it waits on the QR. "Revisit later if it becomes a problem" sounds good!